avocado/utils/astring.py

```diff
@@ -306,21 +310,25 @@ def is_text(data):
     return isinstance(data, str)
 
 
-def to_text(data, encoding=None):
+def to_text(data, encoding=ENCODING):
     """
-    Convert data to text
+    Convert anything to text decoded text
 
-    Action is only taken if data is "bytes", in which case it's
-    decoded into the given encoding and should produce a type that
-    passes the is_text() check.
+    When the data is bytes, it's decoded. When it's not of string types
+    it's re-formatted into text and returned. Otherwise (it's string)
+    it's returned unchanged.
 
     :param data: data to be transformed into text
     :type data: either bytes or other data that will be returned
                 unchanged
     """
     if is_bytes(data):
         if encoding is None:
-            return data.decode()
+            encoding = ENCODING
+        return data.decode(encoding)
+    elif not isinstance(data, string_types):
+        if sys.version_info[0] < 3:
```

Comment:
I assume that repr(xx) is proper way to dump objects as string.

Reply:
This function is not supposed to dump the object, but convert it to text. The difference is:

python2:
```python
>>> repr(Exception(u'\u00e1'))
"Exception(u'\\xe1',)"
>>> unicode(Exception(u'\u00e1'))
u'\xe1'
```

python3
```python
>>> repr(Exception(u'\u00e1'))
"Exception('á',)"
>>> str(Exception(u'\u00e1'))
'á'
```

We use this `to_string` for example when displaying exceptions in Avocado to users, which is suppose to only show the message.